plugins/rikki/loungeviews/components/Navigation.php

```diff
@@ -39,13 +39,19 @@ public function init()
     {
         parent::init();
 
-
-        $this->user = Auth::getUser();
         $this->amateurseasons = Season::where('type', 1)->orderBy('created_at','desc')->get();
         $this->divsseasons = Season::where('type', 2)->orderBy('created_at','desc')->get();
+
+        $this->user = Auth::getUser();
         if ($this->user != null) {
             $this->sloth = SlothModel::getFromUser($this->user);
+
+            if ($this->sloth->timezone == '') {
+                $this->sloth->timezone = TimezoneHelper::getTimezone();
+                $this->sloth->save();
+            }
```

Comment:
I put this here because it seems the most likely to do the desired thing consistently